pkg/apps/controller/deployer/deployer_controller.go

```diff
@@ -246,6 +265,8 @@ func (c *DeploymentController) handle(deployment *v1.ReplicationController, will
 		}
 
 	case deployapi.DeploymentStatusComplete:
+		c.expectations.ExpectDeletions(key, 0)
```

Comment:
nit: ExpectDeletionCountForKey() or ExpectNoDeletions(key) for future generations.